src/function/array/array_functions.cpp

```diff
@@ -139,8 +152,20 @@ scalar_func_exec_t getScalarExecFunc(LogicalType type) {
 template<typename OPERATION>
 std::unique_ptr<FunctionBindData> arrayTemplateBindFunc(std::string functionName,
     const binder::expression_vector& arguments, Function* function) {
-    const auto& leftType = arguments[0]->dataType;
-    const auto& rightType = arguments[1]->dataType;
+    auto leftType = arguments[0]->dataType.copy();
+    auto rightType = arguments[1]->dataType.copy(); // we need left and right to be mutable
+    if (arguments[0]->expressionType == ExpressionType::LITERAL &&
+        leftType.getLogicalTypeID() == LogicalTypeID::LIST) {
+        auto numChildren =
+            arguments[0]->constPtrCast<binder::LiteralExpression>()->getValue().getChildrenSize();
+        leftType = LogicalType::ARRAY(ListType::getChildType(leftType).copy(), numChildren);
+    }
+    if (arguments[1]->expressionType == ExpressionType::LITERAL &&
+        rightType.getLogicalTypeID() == LogicalTypeID::LIST) {
+        auto numChildren =
+            arguments[1]->constPtrCast<binder::LiteralExpression>()->getValue().getChildrenSize();
+        rightType = LogicalType::ARRAY(ListType::getChildType(rightType).copy(), numChildren);
+    }
```

Comment:
Apparently this is duplicated code with l17 -l30. Let's refactor them into a function

Reply:
sounds good